cvat-ui/src/components/annotation-page/appearance-block.tsx

```diff
@@ -144,8 +151,11 @@ function AppearanceBlock(props: Props): JSX.Element {
         changeShapesOutlinedBorders,
         changeShowBitmap,
         changeShowProjections,
+        canvasInstance,
     } = props;
 
+    const is2D = canvasInstance instanceof Canvas;
```

Comment:
Getting by instances is faster, so implemented as such

Reply:
Why is it faster? Did you conduct any experiments? It makes the code looking more complex and less obvious and it is why better to get dimension from the storage.

Reply:
And what is more we do need (and do pass) canvasInstance in some components only for the purpose to make this instanceof comparison. I think this is very controversial solution.

Reply:
In this case there are only 2 possibilities of task, 2D and 3D which constitutes canvas and canvas3d.

These are easily accessible data structures across the code as we have defined its type as well, thus getting 'instance of' is a quick access , also in chrome engine the instance of has no effects on performance as well.

Reply:
@manasars 

The question is not about performance. I believe the both ways are fast enough, but the question is about code readability and clarity.
From this POV it is better to get dimension from storage instead of trying getting it from instanceof operator, which is not very clear. 

In other ways appearance block (and other components in the review) should know NOTHING about the canvas, we do not want this API depend on canvas. This is extra, unnecessary link in the code.